projects/libs/core-library-typescript/test/virtual-machine/event-counter-smart-contract.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { expect } from "chai";
+import "mocha";
+
+import { StateCache } from "../../src/virtual-machine/state-cache";
+import { BaseContractStateAccessor } from "../../src/virtual-machine/contract-state-accessor";
+import EventCounterContract from "../../src/virtual-machine/hard-coded-contracts/registry/event-counter-contract";
+import BaseSmartContract from "../../src/virtual-machine/hard-coded-contracts/base-smart-contact";
+import { types } from "../../src";
+import { Address } from "../../src/common-library/address";
+import { createHash } from "crypto";
+
+export default class ContractStateMemCacheAccessor extends BaseContractStateAccessor {
+  lastBlockId: number;
+  stateCache: StateCache;
+
+  constructor(contractAddress: Buffer, stateCache: StateCache) {
+    super(contractAddress);
+
+    this.stateCache = stateCache;
+  }
+
+  async load(key: string) {
+    return this.stateCache.get({ contractAddress: this.contractAddress, key });
+  }
+
+  async store(key: string, value: string) {
+    this.stateCache.set({ contractAddress: this.contractAddress, key }, value);
+  }
+}
+
+const CONTRACT_ADDRESS = Address.createContractAddress("event-counter-contract").toBuffer();
+const ALICE_ADDRESS = new Address(createHash("sha256").update("Alice").digest()).toBase58();
+const BOB_ADDRESS = new Address(createHash("sha256").update("Bob").digest()).toBase58();
+
+let adapter: BaseContractStateAccessor;
+let aliceContract: BaseSmartContract;
+let bobContract: BaseSmartContract;
+
+describe("event counter contract ", () => {
+  beforeEach(() => {
+    adapter = new ContractStateMemCacheAccessor(CONTRACT_ADDRESS, new StateCache());
+    aliceContract = new EventCounterContract(ALICE_ADDRESS, adapter);
+    bobContract = new EventCounterContract(BOB_ADDRESS, adapter);
+  });
+
+
```

Comment:
add a test for when you do not send a string and make sure it does not update the store. (2 as any) will override typescript checks and allow an int afaik

Reply:
Done.